pkg/nbictl/nbictl_data.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 EdgeXR, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nbictl provides client functions for accessing the NBI
+// (north bound interface)
+package nbictl
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"github.com/edgexr/edge-cloud-platform/api/nbi"
+)
+
+type AppInst struct {
+	AppID    nbi.AppId    `json:"appid"`
+	AppZones nbi.AppZones `json:"appzones"`
+}
+
+type ApplyData struct {
+	Apps     []nbi.AppManifest `json:"apps,omitempty"`
+	AppInsts []AppInst         `json:"appinsts,omitempty"`
+}
+
+type GetData struct {
+	Apps     []nbi.AppManifest     `json:"apps,omitempty"`
+	AppInsts []nbi.AppInstanceInfo `json:"appinsts,omitempty"`
+	Zones    []nbi.EdgeCloudZone   `json:"zones,omitempty"`
+}
+
+type APIErr struct {
+	Desc   string
+	Status int
+	Err    error
+}
+
+func getURL(addr string) string {
+	return addr + "/edge-application-management/vwip"
```

Comment:
Since this is used in several places, might make sense to create a constant for this.